components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/workflow/APIStateChangeWorkflow.java

```diff
@@ -63,13 +65,15 @@
     private ApiDAO apiDAO;
     private APISubscriptionDAO apiSubscriptionDAO;
     private APILifecycleManager apiLifecycleManager;
+    private APIGateway apiGateway;
 
     public APIStateChangeWorkflow(ApiDAO apiDAO, APISubscriptionDAO apiSubscriptionDAO, WorkflowDAO workflowDAO,
-                                  APILifecycleManager apiLifecycleManager) {
+                                  APILifecycleManager apiLifecycleManager, APIGateway apiGateway) {
```

Comment:
API Manager 3.0 is under development with C5. We haven't done a major release yet and major development is still on going. Since these classes are still subject to major changes, we are continuously doing refactoring and heavy development. After major release we will adhere to the versioning.